File: radis/src/main/java/fr/geobert/radis/tools/TargetSumWatcher.java
package fr.geobert.radis.tools;

import android.text.Editable;
import android.widget.EditText;

public class TargetSumWatcher extends CorrectCommaWatcher {
    private final UpdateDisplayInterface updateItf;

    public TargetSumWatcher(char localeComma, EditText w, UpdateDisplayInterface update) {
        super(localeComma, w);
        this.updateItf = update;
    }

    @Override
    public void afterTextChanged(Editable s) {
        correctComma(s);
        updateItf.updateDisplay(null);
    }
}

File: app/src/main/kotlin/fr/geobert/radis/tools/TargetSumWatcher.kt
package fr.geobert.radis.tools

import android.text.Editable
import android.widget.EditText
import fr.geobert.radis.ui.CheckingOpDashboard

public class TargetSumWatcher(localeComma: Char, localeGroupSep: Char, w: EditText, val checkingDashboard: CheckingOpDashboard) : CorrectCommaWatcher(localeComma, localeGroupSep, w) {
    override fun afterTextChanged(s: Editable) {
        correctComma(s)
        checkingDashboard.updateDisplay()
    }
}
